src/NetAnalyzers/CSharp/Microsoft.NetCore.Analyzers/Runtime/CSharpForwardCancellationTokenToInvocationsFixer.TypeNameVisitor.cs

```diff
@@ -0,0 +1,233 @@
+
+namespace Microsoft.NetCore.CSharp.Analyzers.Runtime
+{
+    public sealed partial class CSharpForwardCancellationTokenToInvocationsFixer
+    {
+        private class TypeNameVisitor : SymbolVisitor<TypeSyntax>
+        {
+            public static TypeSyntax GetTypeSyntaxForSymbol(INamespaceOrTypeSymbol symbol)
+            {
+                return symbol.Accept(new TypeNameVisitor()).WithAdditionalAnnotations(Simplifier.Annotation);
+            }
+
+            public override TypeSyntax DefaultVisit(ISymbol symbol)
+                => throw new NotImplementedException();
+
+            public override TypeSyntax VisitAlias(IAliasSymbol symbol)
+            {
+                return AddInformationTo(ToIdentifierName(symbol.Name));
+            }
+
+            public override TypeSyntax VisitDynamicType(IDynamicTypeSymbol symbol)
+            {
+                return AddInformationTo(IdentifierName("dynamic"));
+            }
+
+            public override TypeSyntax VisitNamedType(INamedTypeSymbol symbol)
+            {
+                if (TryCreateNativeIntegerType(symbol, out var typeSyntax))
+                    return typeSyntax;
+
+                typeSyntax = CreateSimpleTypeSyntax(symbol);
+                if (!(typeSyntax is SimpleNameSyntax))
+                    return typeSyntax;
+
+                var simpleNameSyntax = (SimpleNameSyntax)typeSyntax;
+                if (symbol.ContainingType is not null)
+                {
+                    if (symbol.ContainingType.TypeKind != TypeKind.Submission)
+                    {
+                        var containingTypeSyntax = symbol.ContainingType.Accept(this);
+                        if (containingTypeSyntax is NameSyntax name)
+                        {
+                            typeSyntax = AddInformationTo(
+                                QualifiedName(name, simpleNameSyntax));
+                        }
+                        else
+                        {
+                            typeSyntax = AddInformationTo(simpleNameSyntax);
+                        }
+                    }
+                }
+                else if (symbol.ContainingNamespace is not null)
+                {
+                    if (symbol.ContainingNamespace.IsGlobalNamespace)
+                    {
+                        if (symbol.TypeKind != TypeKind.Error)
+                        {
+                            typeSyntax = AddGlobalAlias(simpleNameSyntax);
+                        }
+                    }
+                    else
+                    {
+                        var container = symbol.ContainingNamespace.Accept(this)!;
+                        typeSyntax = AddInformationTo(QualifiedName(
+                            (NameSyntax)container,
+                            simpleNameSyntax));
+                    }
+                }
+
+                if (symbol.NullableAnnotation() == NullableAnnotation.Annotated &&
```

Comment:
Is it possible for `symbol.NullableAnnotation()` to return `Annotated` for value types?